kit/scrolls/src/Lists/UIMasonryList.tsx

```diff
@@ -297,16 +323,34 @@ function useVirtualization<Item>(
     React.useEffect(() => {
         /**
          * So basically when we render items for the first time,
+         * or when the data has been updated,
          * we have to calculate items that're in virtual window
          */
         const bottom = binarySearch(
-            initialWindowHeight + virtualWindowSizeRatio * initialWindowHeight,
+            initialWindowHeight +
+                virtualWindowSizeRatio * initialWindowHeight +
+                currentScrollValue.current,
             cellsIndexes,
         );
-        manageBottomCells(lastBottomCellIndex.current, bottom, show, hide);
+        const top = binarySearch(
+            currentScrollValue.current -
+                virtualWindowSizeRatio * initialWindowHeight -
+                maxItemHeight,
+            cellsIndexes,
+        );
+
+        manageBottomCells(top, bottom, show, hide);
```

Comment:
Not sure I understand the logic now.... Befoe we've compared current and prev bottom positions, now it's something new and the `top` item shouldn't be connected to the `bottom` management... Or I'm missing something. Can you clarify, please?

Reply:
There was a bug, when data was updated 
![image](https://user-images.githubusercontent.com/36547799/145026529-5865da96-caa9-436a-89ef-7f47a4a2951d.png)

Reply:
I actually that the whole effect isn't correct anymore. Before, as it was for first render only, I made an assumption that only bottom views should be managed (and actually this optimisation still can be applied if current scroll value is 0 or small enough (maybe less than virtual window?), but now we should do it "right" way:
```js
hide(0, top - 1)
show(top, bottom)
hide(bottom + 1, last)
```
PS. Not sure how to properly apply +-1